### ISSUE TITLE
Keeps loading 

### ISSUE THREAD

**Issue by login22rvp** [jonjomckay/fritter#95]
I just updated it to version 2.5.
It keeps loading.
Also after clicking on any subscription it gives following error. 
![Screenshot_2021-05-17-07-32-20-201_com jonjomckay fritter](https://user-images.githubusercontent.com/83332727/118423243-33a13880-b6e2-11eb-979f-28eb44080bef.jpg)

**Comment by login22rvp** [comment]
I think when I click on "All" then it keeps loading but if it click on groups then it loads properly.

**Comment by jonjomckay** [comment]
@login22rvp how many accounts are you subscribed to in total? There's a bug currently where having a few hundred subscriptions ends up erroring, which I plan to look into this week (#82)

**Comment by bibin-v2020** [comment]
Same for me. I've updated to v2.6. The 'All' is keep loading. I've only 14 subscriptions..

**Comment by eredrolo** [comment]
I have the same problem since I updated to 2.5.0. I have 56 subscriptions. Only the All group keeps loading and others are working fine.

But if I create a new group and add all the subscriptions in it then it works fine. Everyone with this problem can use this as a workaround.

**Comment by RequestPrivacy** [comment]
Same here after 2.5.0 update, 98 subscriptions. Each one on his own is loading fine.

**Comment by bibin-v2020** [comment]
On v2.6.1, i'm getting different error
![Screenshot_2021-05-21-22-30-11-06](https://user-images.githubusercontent.com/84482532/119173054-53d74b80-ba84-11eb-9cab-77fc22d60fd2.jpg)

**Comment by jonjomckay** [comment]
@bibin-v2020 curious. Can you try force stopping the app, and starting it again? It's almost like the database migration didn't run on your device...

**Comment by bibin-v2020** [comment]
Tried that already..cleared all data (fresh install) as well and ended up in some code error now..
![Uploading Screenshot_2021-05-21-22-43-26-06.jpg…]()

**Comment by jonjomckay** [comment]
It looks like the database migrations didn't run - can you try with the latest build from the `master` branch? I've hopefully fixed it there. You can download it [here](https://github.com/jonjomckay/fritter/actions/runs/866942151)

**Comment by bibin-v2020** [comment]
> It looks like the database migrations didn't run - can you try with the latest build from the `master` branch? I've hopefully fixed it there. You can download it [here](https://github.com/jonjomckay/fritter/actions/runs/866942151)

Thanks!! I installed the apk from master branch but the app didn't even open, just a black page
![Uploading Screenshot_2021-05-23-08-00-24-18.jpg…]()

**Comment by jonjomckay** [comment]
This is very strange... I know it might be a big ask, but are you able to get me the logcat output? There are a few apps on F-Droid that should be able to get it for you

**Comment by kajdo** [comment]
Have the same issue ...

Tried 2.6 and 2.5
Device oneplus5t

Assumed it might be blokada blocking something, but also with disabled vpn its the same

Nr of subs might not be the cause (tested with 2 random subs)

Tried clean install, so i dont think a migration is the cause

**Comment by RequestPrivacy** [comment]
I just updated Fritter to newest F-Droid version (2.6.0). No change, it still keeps loading the group feed endlessly. But I noticed a first workaround for me:

1. On the loading page/group feed go to the settings in the upper right corner
2. Remove both filter checkmarks ("Include replies" & "Include retweets")
3. Exit to subscription overview page
4. Once again load group feed

> Same here after 2.5.0 update, 98 subscriptions. Each one on his own is loading fine.

**Comment by WindHun7er** [comment]
On my side I re-installed every version from 2.5.0 to 2.6.1 (official version). Versions 2.5.0 and 2.6.0 have the same problem and indeed the loading is infinite and the wheel turns indefinitely (it is not a question here of the number of subscriptions as in the issue #82 I opened previously, I took care here to add only a few dozen subscriptions). The surprising thing is that, on the one hand, there doesn't seem to be any error or crash in the logcat (but I may have missed it...). Secondly, from the start the whole interface does not load compared to version 2.6.1:

v2.5.0 et 2.6.0 :

![Screenshot_20210523-182951_Fritter](https://user-images.githubusercontent.com/80457123/119269073-2ad6c800-bbf6-11eb-995d-3fb14a10d02a.jpg)

v2.6.1 (before complete loading) :

![Screenshot_20210523-183024_Fritter](https://user-images.githubusercontent.com/80457123/119269074-2c07f500-bbf6-11eb-9b1e-98a802a96a5a.jpg)


Concerning version 2.6.1 I have no problem and everything works normally again (except the number of subscriptions notified in the issue #82 obviously...)

**Comment by WindHun7er** [comment]
Finally I found this (Don't know if it's related) :

```
2021-05-23 18:53:35.374 17642-18312/? E/flutter: [ERROR:flutter/lib/ui/ui_dart_state.cc(186)] Unhandled Exception: Connection closed while receiving data
    #0      IOClient.send.<anonymous closure> (package:http/src/io_client.dart:49)
    #1      _invokeErrorHandler (dart:async/async_error.dart:45)
    #2      _HandleErrorStream._handleError (dart:async/stream_pipe.dart:272)
    #3      _ForwardingStreamSubscription._handleError (dart:async/stream_pipe.dart:157)
    #4      _HttpClientResponse.listen.<anonymous closure> (dart:_http/http_impl.dart:441)
    #5      _rootRunBinary (dart:async/zone.dart:1378)
    #6      _CustomZone.runBinary (dart:async/zone.dart:1272)
    #7      _CustomZone.runBinaryGuarded (dart:async/zone.dart:1178)
    #8      _BufferingStreamSubscription._sendError.sendError (dart:async/stream_impl.dart:360)
    #9      _BufferingStreamSubscription._sendError (dart:async/stream_impl.dart:378)
    #10     _BufferingStreamSubscription._addError (dart:async/stream_impl.dart:280)
    #11     _SinkTransformerStreamSubscription._addError (dart:async/stream_transformers.dart:75)
    #12     _EventSinkWrapper.addError (dart:async/stream_transformers.dart:17)
    #13     _ConverterStreamEventSink.addError (dart:convert/chunked_conversion.dart:77)
    #14     _SinkTransformerStreamSubscription._handleError (dart:async/stream_transformers.dart:119)
    #15     _rootRunBinary (dart:async/zone.dart:1378)
    #16     _CustomZone.runBinary (dart:async/zone.dart:1272)
    #17     _CustomZone.runBinaryGuarded (dart:async/zone.dart:1178)
    #18     _BufferingStreamSubscription._sendError.sendError (dart:async/stream_impl.dart:360)
    #19     _BufferingStreamSubscription._sendError (dart:async/stream_impl.dart:378)
    #20     _BufferingStreamSubscription._addError (dart:async/stream_impl.dart:280)
    #21     _SinkTransformerStreamSubscription._addError (dart:async/stream_transformers.dart:75)
    #22     _EventSinkWrapper.addError (dart:async/stream_transformers.dart:17)
    #23     _ConverterStreamEventSink.addError (dart:convert/chunked_conversion.dart:77)
    #24     _SinkTransformerStreamSubscription._handleError (dart:async/stream_transformers.dart:119)
    #25     _rootRunBinary (dart:async/zone.dart:1378)
    #26     _CustomZone.runBinary (dart:async/zone.dart:1272)
    #27     _CustomZone.runBinaryGuarded (dart:async/zone.dart:1178)
    #28     _BufferingStreamSubscription._sendError.sendError (dart:async/stream_impl.dart:360)
    #29     _BufferingStreamSubscription._sendError (dart:async/stream_impl.dart:378)
    #30     _BufferingStreamSubscription._addError (dart:async/stream_impl.dart:280)
    #31     _ForwardingStreamSubscription._addError (dart:async/stream_pipe.dart:128)
    #32     _addErrorWithReplacement (dart:async/stream_pipe.dart:176)
    #33     _HandleErrorStream._handleError (dart:async/stream_pipe.dart:277)
    #34     _ForwardingStreamSubscription._handleError (dart:async/stream_pipe.dart:157)
    #35     _rootRunBinary (dart:async/zone.dart:1378)
    #36     _CustomZone.runBinary (dart:async/zone.dart:1272)
    #37     _CustomZone.runBinaryGuarded (dart:async/zone.dart:1178)
    #38     _BufferingStreamSubscription._sendError.sendError (dart:async/stream_impl.dart:360)
    #39     _BufferingStreamSubscription._sendError (dart:async/stream_impl.dart:378)
    #40     _BufferingStreamSubscription._addError (dart:async/stream_impl.dart:280)
    #41     _SyncStreamControllerDispatch._sendError (dart:async/stream_controller.dart:737)
    #42     _StreamController._addError (dart:async/stream_controller.dart:615)
    #43     _StreamController.addError (dart:async/stream_controller.dart:569)
    #44     _HttpParser._reportBodyError (dart:_http/http_parser.dart:1163)
    #45     _HttpParser._onDone (dart:_http/http_parser.dart:870)
    #46     _rootRun (dart:async/zone.dart:1346)
    #47     _CustomZone.run (dart:async/zone.dart:1258)
    #48     _CustomZone.runGuarded (dart:async/zone.dart:1162)
    #49     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:394)
    #50     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:404)
    #51     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:291)
2021-05-23 18:53:35.374 17642-18312/? E/flutter: #52     _SyncStreamControllerDispatch._sendDone (dart:async/stream_controller.dart:741)
    #53     _StreamController._closeUnchecked (dart:async/stream_controller.dart:596)
    #54     _StreamController.close (dart:async/stream_controller.dart:589)
    #55     _Socket._onData (dart:io-patch/socket_patch.dart:2151)
    #56     _rootRunUnary (dart:async/zone.dart:1362)
    #57     _CustomZone.runUnary (dart:async/zone.dart:1265)
    #58     _CustomZone.runUnaryGuarded (dart:async/zone.dart:1170)
    #59     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341)
    #60     _BufferingStreamSubscription._add (dart:async/stream_impl.dart:271)
    #61     _SyncStreamControllerDispatch._sendData (dart:async/stream_controller.dart:733)
    #62     _StreamController._add (dart:async/stream_controller.dart:607)
    #63     _StreamController.add (dart:async/stream_controller.dart:554)
    #64     _RawSecureSocket._closeHandler (dart:io/secure_socket.dart:773)
    #65     _RawSecureSocket._tryFilter (dart:io/secure_socket.dart:904)
    <asynchronous suspension>
```

**Comment by jonjomckay** [comment]
Thanks all for your debugging help - I think I've figure out what's going on. The migration I added in v2.6.1 to fix this issue relied on a feature of SQLite that is only supported on newer devices (`ON CONFLICT`), so I've replaced it with a more compatible way of doing the same thing and released it as v2.6.2.

It should be ready to download within the next few minutes, and hopefully should fix the problem 😅 

The next major version (v2.7.0) will focus on better error handling, and will include a way for users to submit their logs from inside the app, so these kinds of issues can hopefully be solved more easily!

**Comment by login22rvp** [comment]
Awesome

**Comment by bibin-v2020** [comment]
Thanks

**Comment by H2172** [comment]
@jonjomckay just to confirm 2.6.2 fixed the issue for me (Android 8.1.0). Thanks a million!

**Comment by jonjomckay** [comment]
Thanks all for confirming this was fixed in v2.6.2!
